### DISCUSSION TITLE
Sorting PR's checks by their status

### DISCUSSION THREAD

**Discussion by alcuadrado** [community/community#7885] [Pull Requests]
I maintain a repository which uses Github Actions as its CI solution. Some Pull Requests run over 80 jobs. The checks lists in PRs are not sorted in any useful way, so when a check fails I have to manually search for it. I'd be really happy if you sorted checks according to their status. Probably something like: failed, running, successful.

**Comment by jimtng** [comment]
+1

I was just about to make a similar suggestion / feature request

**Comment by pahwaranger** [comment]
sort by status first and then alphabetically within each status group would be amazing

**Comment by skjnldsv** [comment]
Made a user Script https://gist.github.com/skjnldsv/eb7b894ae996affde4a7d0e00e0d80a1

**Comment by lbergelson** [comment]
This is the sort of quality of life feature I wish would be implemented instead of sitting for years.

**Comment by ospikovets** [comment]
I would also appreciate to have a way for sorting and, maybe, grouping PR checks.

**Comment by ShuttleSantalex** [comment]
This would be great!

**Comment by ligurio** [comment]
Seems the problem is Firefox-specific and should be fixed in Firefox 119, see https://github.com/refined-github/refined-github/issues/5797#issuecomment-1727140020

**Comment by saiichihashimoto** [comment]
797 days "Unanswered" on a god-tier suggestion with a trivial-time implementation

**Comment by dibstern-work** [comment]
Would love to see this

**Comment by cgruber** [comment]
Would also love to see this - especially bubbling up "Required" checks to the top, with Failing ones prioritized as well.

**Comment by JaeceeNaylor** [comment]
please do this

**Comment by turtleyacht** [comment]
If a workflow is pinned in the Actions tab, then in the pull request, maybe those could stay at the top of the workflow line items (before or after [bubbled Required checks](https://github.com/orgs/community/discussions/7885#discussioncomment-11080431)).

Although--in hindsight--external workflow results like Chromatic could not be pinned in Actions, since they swoop in after. Then there would have to be some way to configure preferences per-user.

Or, if the triggering workflow is pinned, then subsequent 3rd-party workflow results are brought to the top.

**Comment by coderjinhui** [comment]
year 2021 already gone......

**Comment by turtleyacht** [comment]
Noticed recently workflows are now sorted into sections in this order: failing, pending, in-progress, skipped, and successful.

Within each section, workflows are alphabetized per [comment above](https://github.com/orgs/community/discussions/7885#discussioncomment-2384320).

**Comment by seanturner-zh** [comment]
yes, now they just need to do it for the failure email...
